test/e2e/framework/framework.go

```diff
@@ -155,6 +155,27 @@ func (f *Framework) BeforeEach() {
 	if f.ClientSet == nil {
 		By("Creating a kubernetes client")
 		config, err := LoadConfig()
+		testDesc := CurrentGinkgoTestDescription()
+		switch {
+		case len(testDesc.ComponentTexts) > 0:
+			fileName := testDesc.FileName
+			// An attempt at removing $HOME/go/src prefix
+			src_idx := strings.Index(fileName, "k8s.io/kubernetes/test/e2e")
+			if src_idx != -1 {
+				fileName = testDesc.FileName[src_idx:]
+			}
+			// in case of fileName:LineNumber ~ test/e2e/framework/framework.go:757
+			// it's hitting the ConformanceIt() wrapper....
+			// We'll need a work around to find conformance test sources
+			componentTexts := strings.Join(testDesc.ComponentTexts, " ")
+			config.UserAgent = fmt.Sprintf(
+				"%v -- %v:%v -- %v",
+				rest.DefaultKubernetesUserAgent(),
+				fileName,
+				testDesc.LineNumber,
```

Comment:
I'm not sure how interesting or useful it is to go as low level as file:line if we have access to the test description